pyxem/signals/differential_phase_contrast.py

```diff
@@ -271,6 +271,138 @@ def get_magnitude_signal(self, autolim=True, autolim_sigma=4):
+        imaging using hard x-rays. Optics Express, 15(3), p.1175. (Equation 4)
+
+        .. [2] Arnison, M., Larkin, K., Sheppard, C., Smith, N. and
+        Cogswell, C., 2004. Linear phase imaging using differential
+        interference contrast microscopy. Journal of Microscopy, 214(1),
+        pp.7-12. (Equation 6)
+
+        .. [3] Frankot, R. and Chellappa, R., 1988. A method for enforcing
+        integrability in shape from shading algorithms.
+        IEEE Transactions on Pattern Analysis and Machine Intelligence,
+        10(4), pp.439-451. (Equation 21)
+
+        Examples
+        --------
+        >>> from pyxem.dummy_data import dummy_data
+        >>> s = dummy_data.get_square_dpc_signal()
+        >>> s_phase = s.phase_retrieval()
+        >>> s_phase.plot()
+        """
+
+        method = method.lower()
+        if method not in ("kottler", "arnison", "frankot"):
+            raise ValueError(
+                "Method '{}' not recognised. 'kottler', 'arnison'"
+                " and 'frankot' are available.".format(method)
+            )
+
+        # get x and y phase gradient
+        dx = self.inav[0].data
+        dy = self.inav[1].data
+
+        # attempt to reduce boundary effect
+        if mirroring:
+            Ax = dx
+            Bx = np.flip(dx, axis=1)
+            Cx = np.flip(dx, axis=0)
+            Dx = np.flip(dx)
+
+            Ay = dy
+            By = np.flip(dy, axis=1)
+            Cy = np.flip(dy, axis=0)
+            Dy = np.flip(dy)
+
+            # the -ve depends on the direction of derivatives
+            if not mirror_flip:
+                dx = np.bmat([[Ax, -Bx], [Cx, -Dx]]).A
+                dy = np.bmat([[Ay, By], [-Cy, -Dy]]).A
+            else:
+                dx = np.bmat([[Ax, Bx], [-Cx, -Dx]]).A
+                dy = np.bmat([[Ay, -By], [Cy, -Dy]]).A
+
+        nc, nr = dx.shape[1], dx.shape[0]
+
+        # get scan step size
+        calX = np.diff(self.axes_manager.signal_axes[0].axis).mean()
+        calY = np.diff(self.axes_manager.signal_axes[1].axis).mean()
+
+        # construct Fourier-space grids
+        kx = (2 * np.pi) * np.fft.fftshift(np.fft.fftfreq(nc))
+        ky = (2 * np.pi) * np.fft.fftshift(np.fft.fftfreq(nr))
+        kx_grid, ky_grid = np.meshgrid(kx, ky)
+
+        if method == "kottler":
+            gxy = dx + 1j * dy
+            numerator = np.fft.fftshift(np.fft.fft2(gxy))
+            denominator = 2 * np.pi * 1j * (kx_grid + 1j * ky_grid)
+        elif method == "arnison":
+            gxy = dx + 1j * dy
+            numerator = np.fft.fftshift(np.fft.fft2(gxy))
+            denominator = 2j * (
+                np.sin(2 * np.pi * calX * kx_grid)
+                + 1j * np.sin(2 * np.pi * calY * ky_grid)
+            )
+        elif method == "frankot":
+            kx_grid /= calX
+            ky_grid /= calY
+            fx = np.fft.fftshift(np.fft.fft2(dx))
+            fy = np.fft.fftshift(np.fft.fft2(dy))
+            wx, wy = 0.5, 0.5
```

Comment:
I skimmed through the referenced paper, but I couldn't find that `wx` and `wy` should be 0.5.

Reply:
Yes the paper doesn't have them, `wx` and `wy` are the weights for the phase grad in x and y respectively, so if someone decides the phase grad in x is somehow more important, it can be set as `wx=0.8` and `wy=0.2` for example. I included it  during testing but just hardcoded to `0.5` as don't think it will be useful.

I incline to remove `wx` and `wy`, but if someone finds this might be useful, `wx` can be set via argument.